torch/_functorch/aot_autograd.py

```diff
@@ -1419,7 +1426,7 @@ def view_avoid_dupes_with_primals(t):
         f_input_tangents = [
             inp
             for inp, info in zip(flat_f_args, input_info)
-            if info.mutates_data and info.requires_grad
+            if info.mutates_data and info.requires_grad and not info.mutations_hidden_from_autograd
```

Comment:
fwiw (in case we merge conflict), I ended up changing this function in my PR ([link](https://github.com/pytorch/pytorch/pull/113584/files#diff-df954bbf954d2dcb81f687876053267ffa4ddb36ed86b7d2bd76319ff2b94416R1466)) to basically re-use `_get_mutation_type`